### PR TITLE
Merge the new identity label module into the main branch

### PR DESCRIPTION
**Pull request by thysw95**
Integration of the new identity label module into the DBM

## Summary by Sourcery

Integrate a new identity label module into the system, providing CRUD operations for identity labels and enhancing database schema with a new 'identity_labels' table. Refactor database initialization to improve maintainability.

New Features:
- Introduce a new identity label module that allows creating, retrieving, updating, and deleting labels for identities within communities.

Enhancements:
- Add a new database table 'identity_labels' to store labels associated with identities and communities.
- Refactor database initialization logic into a dedicated function 'initializeDB' to improve code organization.

### CODE REVIEW

File: models/db.py [hunk 1]
```diff
@@ -166,6 +166,12 @@
                 Field('community_name'),
                 Field('community_description'))
 
+# Define a table that keeps track of labels for identities in a community
+db.define_table('identity_labels',
+                Field('identity_id', db.identities),
+                Field('community_id', db.communities),
```
[the remark this answers is not in the document]
Comment (thysw95) [reply models/db.py:172]: This is a good suggestion. However, in the current context, each individual in a community can have multiple labels assigned to them, so this is not necessary

File: controllers/identity_label.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+# try something like
+def index(): return dict(message="hello from identity_label.py")
+
+# Function to create a new identity label for a given identity_name and community_name.
+# Throws an error if no identity_name or community_name is given, or the identity label already exists.
+# Also throws an error if the identity or the community does not exist.
+def create():
+    # Define the payload.
+    payload = request.body.read()
```
[the remark this answers is not in the document]
Comment (thysw95) [reply controllers/identity_label.py]: Ok is implemented!

File: controllers/identity_label.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+import json
```
Comment (PenguinzPlays) [inline controllers/identity_label.py:2]: Do we need all of the json library?

File: controllers/identity_label.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    return payload
+
+# Function to create a new identity label for a given identity_name and community_name.
+# Throws an error if no identity_name or community_name is given, or the identity label already exists.
+# Also throws an error if the identity or the community does not exist.
+def create():
+    # Define the payload.
+    try:
+        payload = parse_payload()
+    except ValueError as e:
```
Comment (PenguinzPlays) [inline controllers/identity_label.py:22]: avoid single unreadable character variables, instead use something like err

**Review by PenguinzPlays — APPROVED** [review]
Some bad habits showing in here, but fine with it for now.

**Pull request merged**